examples/rates/retrieve_rate.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/EasyPost/easypost-go"
+	"os"
+)
+
+func main() {
+	apiKey := os.Getenv("EASYPOST_API_KEY")
+	if apiKey == "" {
+		fmt.Fprintln(os.Stderr, "missing API key")
+		os.Exit(1)
+		return
+	}
+	client := easypost.New(apiKey)
+
+	// Retrieve a rate
+	rate, err := client.GetRate("rate_123")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error retrieving rate:", err)
+		os.Exit(1)
+		return
+	}
+
+	prettyJSON, err := json.MarshalIndent(rate, "", "    ")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error creating JSON:", err)
+	}
```

Comment:
Things would appear to be in an error state at this point, but this just continues on, no exit or return.

Reply:
Not sure I follow, I ran this and it printed the rate object in JSON.

Reply:
If it can't marshal the struct into formatted JSON, it emits the error, and then continues on to print the value of prettyJSON.  This isn't strictly a problem and is probably unlikely, since prettyJSON should be empty, but it is of questionable style to continue on in the event of an error like this: it makes it look like you missed something.

Reply:
Oh I see, it's missing:

```go
os.Exit(1)
return
```

Added and pushed! Thanks.